### ISSUE TITLE
Back Propagation에 대해 쉽게 설명한다면?

### ISSUE THREAD

**Issue by import-max** [AInsteins/AI-Tech-Interview#32]

**Comment by kimyuuuuuu** [comment]
- Back Propagation(역전파) 알고리즘은 Loss(손실)에 대한 입력값의 기울기(미분값)을 출력층 layer에서부터 계산하여 거꾸로 전파시키는 것이다.
- 즉, 내가 뽑고자 하는 **`target`**값과 실제 모델이 계산한 **`output`**이 얼마나 차이가 나는지 구한 후 그 오차값을 다시 뒤로 전파해가면서 각 노드가 가지고 있는 변수들을 갱신하는 알고리즘.
- 역전파 알고리즘을 이용해 파라미터가 매우 많고 layer가 여러개 있을때 가중치w와 b를 학습시키기 어려웠던 문제를 해결하였다. ->  Gradient descent 방법으로 가중치w와 b를 update시키면서 해결

**Comment by miinkang** [comment]
Back Propagation

- 출력층 Layer에서부터 입력값에 대한 미분값을 계산해 거꾸로 전파시키는 것
- cost를 구하는 최적화 알고리즘으로 chain rule을 이용해 연산함으로써 연산량을 줄일 수 있음

chain rule

- cost의 미분값을 이용해 w를 갱신할 때, cost의 미분값을 구하기 위해 사용함

**Comment by import-max** [comment]
참고 reference

[reference 1](https://box-world.tistory.com/19)
[reference 2] (https://evan-moon.github.io/2018/07/19/deep-learning-backpropagation/)

**Comment by jarammm** [comment]
1. **Loss에 대한 입력값의 기울기(미분값)**를 출력층 layer에서부터 계산하여 **거꾸로 전파** -> **chain rule**이 이용
2. 파라미터가 매우 많고 layer가 여러개 있을때 가중치w와 b를 학습시키기 어려웠던 문제 해결 -> 역전파 알고리즘으로 **각 layer에서 기울기 값을 구하고 그 기울기 값을 이용하여 Gradient descent 방법으로 가중치w와 b를 update시키면서 해결**
